### PR TITLE
Added MaxLines property to TextBlock

### PR DESCRIPTION
**Pull request by jp2masa**
## Changes
- Added `MaxLines` property to `TextBlock`.

### CODE REVIEW

File: tests/Avalonia.Skia.UnitTests/TextLayoutTests.cs [hunk 1]
```diff
@@ -506,6 +506,27 @@ public void Should_Include_Last_Line_When_Constraint_Is_Surpassed(string text, i
             }
         }
 
+        [InlineData("0123456789\r\n0123456789\r\n0123456789", 1, 1)]
```
Comment (Gillibald) [inline tests/Avalonia.Skia.UnitTests/TextLayoutTests.cs:509]: I think some test cases are redundant. You could change the test cases to cover this:

```
[InlineData("0123456789\r\n0123456789\r\n0123456789", 0, 3)]
[InlineData("0123456789\r\n0123456789\r\n0123456789", 4, 3)]
[InlineData("0123456789\r\n0123456789\r\n0123456789", 1, 1)]
```

**Comment by Gillibald** [comment]
Looks good to me in general. I added a comment.

**Review by Gillibald — APPROVED** [review]

File: src/Avalonia.Visuals/Media/TextFormatting/TextLayout.cs [hunk 2]
```diff
@@ -73,6 +77,12 @@ public class TextLayout
         /// </summary>
         public double MaxHeight { get; }
 
+
+        /// <summary>
+        /// Gets the maximum number of text lines.
+        /// </summary>
+        public double MaxLines { get; }
```
Comment (pr8x) [inline src/Avalonia.Visuals/Media/TextFormatting/TextLayout.cs:84]: Why is this a double 😅?

Comment (Gillibald) [reply src/Avalonia.Visuals/Media/TextFormatting/TextLayout.cs:84]: Hmm. Good catch. Didn't realize it is a double. Probably some copy paste error.

Comment (jp2masa) [reply src/Avalonia.Visuals/Media/TextFormatting/TextLayout.cs:84]: Yes, it was a copy paste mistake, should I open a PR to fix that?

Comment (Gillibald) [reply src/Avalonia.Visuals/Media/TextFormatting/TextLayout.cs:84]: Will fix it myself

**Pull request merged**